File: leetcode/75-SortColors/75-SortColors.py
# Last updated: 29/04/2026, 23:13:06
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0
        r = len(nums) - 1
        k = 0
        while k <= r:
            if nums[k] == 0:
                nums[l],nums[k] = nums[k],nums[l]
                l += 1
            elif nums[k] == 2:
                nums[r],nums[k] = nums[k],nums[r]
                r -= 1
                k -= 1
            k += 1

File: leetcode/75-SortColors/75-SortColors.cpp
// Last updated: 29/04/2026, 23:13:10
class Solution {
public:
    
    void swap(int& x, int& y) {
        int temp;
        temp = y;
        y = x;
        x = temp;
    }
    
    void sortColors(vector<int>& nums) {
        int low = 0;
        int med = 0;
        int high = nums.size() - 1;
        while (med <= high) {
            if (nums[med] == 0) {
                swap(nums[med], nums[low]);
                med ++;
                low ++;
            } else if (nums[med] == 1) {
                med ++;
            } else if (nums[med] == 2) {
                swap(nums[med], nums[high]);
                high --;
            }
        }
    }
};
